Libs/Visualization/VTK/Widgets/ctkVTKVolumePropertyWidget.cpp

```diff
@@ -222,14 +222,14 @@ void ctkVTKVolumePropertyWidget::updateFromVolumeProperty()
   if (d->VolumeProperty)
   {
     colorTransferFunction =
-      d->VolumeProperty->GetRGBTransferFunction()->GetSize() ?
-      d->VolumeProperty->GetRGBTransferFunction() : 0;
+      d->VolumeProperty->GetRGBTransferFunction(d->CurrentComponent)->GetSize() ?
```

Comment:
VTK can crash if the component is out of range.

```python
p=vtk.vtkVolumeProperty()
p.GetScalarOpacity(1) # OK
p.GetScalarOpacity(10) # CRASH
```

It could be nice to make things more robust in VTK, but for now let's just make sure that we don't crash the application for unexpected inputs. If the component index is <0 or >=VTK_MAX_VRCOMP (currently 4) then we could disable the widget.